Framework/DataHandling/test/LoadInstrumentTest.h

```diff
@@ -140,17 +128,18 @@ class LoadInstrumentTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(output->getAxis(1)->spectraNo(257), 258);
 
     auto ids_from_map = output->getSpectrum(257).getDetectorIDs();
-    IDetector_const_sptr det_from_ws = output->getDetector(257);
+	const auto &spectrumInfo = output->spectrumInfo();
+    const auto &det_from_ws = spectrumInfo.detector(257);
     TS_ASSERT_EQUALS(ids_from_map.size(), 1);
     TS_ASSERT_EQUALS(*ids_from_map.begin(), 602);
-    TS_ASSERT_EQUALS(det_from_ws->getID(), 602);
+    TS_ASSERT_EQUALS(det_from_ws.getID(), 602);
 
     // also a few tests on the last detector and a test for the one beyond the
     // last
-    boost::shared_ptr<const IDetector> ptrDetLast = i->getDetector(413256);
-    TS_ASSERT_EQUALS(ptrDetLast->getID(), 413256);
-    TS_ASSERT_EQUALS(ptrDetLast->getName(), "pixel");
-    TS_ASSERT_THROWS(i->getDetector(413257), Exception::NotFoundError);
+    const auto &ptrDetLast = detectorInfo.detector(413256);
+    TS_ASSERT_EQUALS(ptrDetLast.getID(), 413256);
+    TS_ASSERT_EQUALS(ptrDetLast.getName(), "pixel");
+    //TS_ASSERT_THROWS_ANYTHING(detectorInfo.detector(413257));
```

Comment:
Can you try `   const auto &ptrDetLast = detectorInfo.detector(detectorInfo.indexOf(413256))`